app/models/sections/info/movementScenario/MovementScenario.scala

```diff
@@ -18,16 +18,26 @@ package models.sections.info.movementScenario
 
 import models.requests.DataRequest
 import models.response.{InvalidDestinationTypeException, InvalidUserTypeException}
+import models.sections.info.movementScenario.MovementScenario.logger
 import models.{Enumerable, UserType, WithName}
 import utils.Logging
 
 sealed trait MovementScenario {
-  def originType(implicit request: DataRequest[_]): OriginType
+  def originType(implicit request: DataRequest[_]): OriginType =
+    (request.isWarehouseKeeper, request.isRegisteredConsignor, request.isCertifiedConsignor) match {
+      case (true, _, _) => OriginType.TaxWarehouse
+      case (_, true, _) => OriginType.Imports
+      case (_, _, true) => OriginType.DutyPaid
+      case _ =>
+        logger.error(s"[getOriginType] invalid UserType for CAM journey: ${request.userTypeFromErn}")
```

Comment:
should this log reference COD instead of CAM? Same question for the exception below

Reply:
Yeah, will get that changed 👍